src/template_items/templateItem.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable, NotFoundException } from "@nestjs/common";
+import { InjectModel } from "@nestjs/mongoose";
+import { Model } from "mongoose";
+import { User } from "src/users/schemas/user.schema";
+import { StatusType, TemplateDocument, TemplateItem } from "./schemas/template.schema";
+import { TemplateItemDto } from "./dto/template.dto";
+import { ObjectId } from "mongoose";
+import { UpdateTemplateDto } from "./dto/updateTemplate.dto";
+import { GetAllTemplatesDto } from "./dto/getAllTemplate.dto";
+
+@Injectable()
+export class TemplateItemService {
+  constructor(
+    @InjectModel(TemplateItem.name) private readonly TemplateModel: Model<TemplateDocument>,
+    @InjectModel(User.name) private readonly userModel: Model<User>,
+  ) {}
+
+  async getAllTemplates(getAllTemplates: GetAllTemplatesDto): Promise<TemplateItem[]> {
+    try {
+      const { user_id, limit = 0, offset = 0 } = getAllTemplates;
+      const optionalQuery: { [key: string]: any } = {};
+
+      if (user_id) optionalQuery.user_id = user_id;
+
+      const response = await this.TemplateModel.find({
+        isDeleted: false,
+        ...optionalQuery,
+      })
+        .sort({ createdAt: -1 })
+        .skip(offset)
+        .limit(limit)
+        .exec();
+
+      if (user_id && response) {
+        return response.filter((res) => res.status !== StatusType.ILLEGAL);
+      } else {
+        return response.filter((res) => res.status === StatusType.PUBLISHED);
+      }
+    } catch (err) {
+      throw new NotFoundException({
+        message: "Something went wrong, pls try again",
```

Comment:
Avoid to use abbreviations "please try again"

Reply:
thank you for your comments, have changed the exception content.